### DISCUSSION TITLE
Mainnet Beta Release 1.3.0beta1

### DISCUSSION THREAD

**Discussion by lk86** [MinaProtocol/mina#10152] [Announcements]
The O(1) Labs team is proud to announce that the 1.3.0 release has graduated to Beta! This 1.3.0beta1 includes a variety of changes back to even before mainnet launch like support for MINA_ environment variables, major changes across all of our testing and infrastructure, archive node optimizations, Rosetta support, and bugfixes and performance improvements across the codebase. Below is a summary of some of the more critical changes (more detailed documentation coming soon).

If you run into any issues, please include `Release: 1.3.0beta1` in the name of any issues you create. Please report any issues you encounter [here](https://github.com/MinaProtocol/mina/issues/new/choose) or join the Discussion below!

### New Features:
- Simpler optimization for staged ledger diff application (and revert buggy PRs from alpha2) #10121
- This version supports both MINA_ environment variables and the old CODA_ prefix, but the 1.3.X will be the last releases to support CODA_. We recommend you take this opportunity to use MINA_ everywhere going forward (example: MINA_PRIVKEY_PASS, MINA_CLIENT_TRUSTLIST, etc.)
- Support for Debian Buster in addition to Ubuntu18.04/ Debian Stretch. Official support for 20.04 is coming soon, hopefully before 1.3.0 stable. To use the new buster-based package or images, replace `stretch` with `buster` in either the docker image tag or the debian repository setup command.
- New command `mina advanced runtime-config` and related `runtimeConfig` graphQL query for returning the configuration the daemon is using #8648
- Upgrade to Ocaml 4.11.2 #8898
- Upgrade to Rust 1.52.1 #9076
- Remove deprecated feature for swapping block producer keys at runtime to simplify daemon startup and fix demo mode #10139

### Archive Node:
- New archive node schema for optimized balance queries #9859 

### Rosetta Support:
- New docker image mina-rosetta-ubuntu which includes the mina daemon, mina-archive, and mina-rosetta along with entrypoint scripts to fill the archive database with blocks back to genesis and start all 3 daemons in tandem.
- Rosetta work has been ongoing for some time, and it has taken 26 pull requests to get all of the features supported and the kinks ironed out, but we are happy with the result. See the [README](https://github.com/MinaProtocol/mina/tree/compatible/src/app/rosetta#readme) for more about working with mina-rosetta.
- Those 26 PRs: 
- #9797 
- #9802 
- #9811 
- #9839 
- #9841 
- #9848 
- #9859 
- #9871 
- #9880 
- #9881 
- #9884 
- #9890 
- #9896 
- #9899 
- #9904 
- #9908 
- #9911 
- #9885 
- #9954 
- #9967 
- #9969 
- #9965 
- #10019 
- #10044 
- #9932 
- #10104

### Complete Changelog since 1.2.2:
The change set is far too large for github to display, but the github comparison still shows the immense scale of 1.3.0 and why we are so excited to get it out into your hands to test.
https://github.com/MinaProtocol/mina/compare/1.2.2...1.3.0beta1
For more information about the changes included in more recent alphas like 1.3.0alpha3, see the [previous release notes](https://github.com/MinaProtocol/mina/releases/tag/1.3.0alpha3)

### Debian Packages:
**Debian Repository:**
First, set up and update the `beta` Debian Repository for your platform (`stretch` or `buster`, ubuntu 20.04 `focal` coming soon)
```
echo "deb [trusted=yes] http://packages.o1test.net stretch beta" | sudo tee /etc/apt/sources.list.d/mina-beta.list
sudo apt-get update
```
Then, install the package(s) that you need:
**Mina Daemon:** `sudo apt-get install -y mina-mainnet=1.3.0beta1-087f715`
**Mina Archive:** `sudo apt-get install -y mina-archive=1.3.0beta1-087f715`
**New Mina Archive Schema:** [create_schema.sql](https://github.com/MinaProtocol/mina/blob/087f71500820c335f1afef9a4bee9a92b5034c70/src/app/archive/create_schema.sql)

### Docker Images:
**Mina Daemon:** `minaprotocol/mina-daemon:1.3.0beta1-087f715-stretch-mainnet`
or for a Debian Buster image: `minaprotocol/mina-daemon:1.3.0beta1-087f715-buster-mainnet`
**Mina Archive:** `minaprotocol/mina-archive:1.3.0beta1-087f715-buster`
**New Mina Archive Schema:** [create_schema.sql](https://github.com/MinaProtocol/mina/blob/087f71500820c335f1afef9a4bee9a92b5034c70/src/app/archive/create_schema.sql)
**Mina Rosetta:** `minaprotocol/mina-rosetta-ubuntu:1.3.0beta1-087f715-stretch`
Do not worry that it is labeled "Stretch", this image is based on Ubuntu 20.04 Focal Fossa.

### Sandbox Node: For testing in an isolated, single-node network without snarks
As of 1.2.0, the demo functionality is built into the standard daemon container, via the environment variable `RUN_DEMO=true`.
Just `docker run -it --name mina-demo -e RUN_DEMO=true minaprotocol/mina-daemon:1.3.0beta1-087f715-buster-devnet` and go!

### Step by Step Guide:
Check out [our documentation](https://docs.minaprotocol.com/en/using-mina/connecting) for complete instructions on using this version to connect to Mainnet. *Docker only* Make sure to run with `--peer-list-url https://storage.googleapis.com/mina-seed-lists/mainnet_seeds.txt`.

If you are running the correct version on the correct network **MAINNET**, `mina client status` will show:
```
Chain id:     5f704cc0c82e0ed70e873f0893d7e06f148524e3f0bdae2afb02e7819a0c24d1
Git SHA-1:    087f71500820c335f1afef9a4bee9a92b5034c70
```

<hr /><em>This discussion was created from the release <a href='https://github.com/MinaProtocol/mina/releases/tag/1.3.0beta1'>Mainnet Beta Release 1.3.0beta1</a>.</em>

**Comment by ghost** [comment]
I've got OOMs on this version. Host with 128 GB RAM...
![image](https://user-images.githubusercontent.com/52304443/152899885-9e3cffdd-032c-43d3-b8ce-9f4fcc1c0e3b.png)
![image](https://user-images.githubusercontent.com/52304443/152900174-9788aba5-a5d7-481e-8727-e1827181ef69.png)


```bash
Scada UUID:

Grouped for:
alertname = HostOomKillDetected

Status: FIRING 🔥

Active Alert List:
  Alert: 
  Current value:Severity: warning
  Active from: 07/02/2022 10:31:57
  
  description: OOM kill detected
  VALUE = 2.6666666666666665
```
[coda_crash_report_2022-02-07_07-31-48.048344.tar.gz](https://github.com/MinaProtocol/mina/files/8013413/coda_crash_report_2022-02-07_07-31-48.048344.tar.gz)
